shell/browser/api/electron_api_desktop_capturer.cc

```diff
@@ -24,12 +25,126 @@
 #include "third_party/webrtc/modules/desktop_capture/desktop_capture_options.h"
 #include "third_party/webrtc/modules/desktop_capture/desktop_capturer.h"
 
+#if defined(USE_OZONE)
+#include "ui/ozone/buildflags.h"
+#if BUILDFLAG(OZONE_PLATFORM_X11)
+#define USE_OZONE_PLATFORM_X11
+#endif
+#endif
+
 #if BUILDFLAG(IS_WIN)
 #include "third_party/webrtc/modules/desktop_capture/win/dxgi_duplicator_controller.h"
 #include "third_party/webrtc/modules/desktop_capture/win/screen_capturer_win_directx.h"
 #include "ui/display/win/display_info.h"
+#elif BUILDFLAG(IS_LINUX)
+#if defined(USE_OZONE_PLATFORM_X11)
+#include "base/logging.h"
+#include "ui/base/x/x11_display_util.h"
+#include "ui/base/x/x11_util.h"
+#include "ui/display/util/display_util.h"
+#include "ui/display/util/edid_parser.h"
```

Comment:
This line is causing a GN error as seen here:
https://app.circleci.com/pipelines/github/electron/electron/53624/workflows/0be2d061-323d-4fbf-a88d-ac157fa91849/jobs/1227792

Reply:
Hm, curious that it doesn't give the error locally for me -- only on macOS & Windows builds? Is that a pre-processor issue?

As for the dependency issue itself, the edid_parser seems to be private to the display_utils "target" (as I understand that GN complaint). I suppose that exposing it by adding it to a header in there would not be ideal, which I guess would mean copying the EDID parser implementation (or just the subset needed here) into the desktop capturer.

I'm away this weekend, but I can have a crack at that next week, if that is indeed the only solution -- unless there's some easier/accepted way of making that class accessible.

Reply:
Oh, I guess the problem is that, as per the docs, GN check doesn't understand the pre-processor; I added `// nogncheck` for the edid_parser include and got rid of the other one (as it is now unneeded).

I see two build failures, but the linux one doesn't seem to be happening near my code (and is, any case, just a comment being added) and the mac build doesn't actually seem to be using the latest version of the code.